var/spack/repos/builtin/packages/singularity-eos/package.py

```diff
@@ -130,25 +115,23 @@ def cmake_args(self):
             self.define_from_variant("SINGULARITY_USE_FORTRAN", "fortran"),
             self.define_from_variant("SINGULARITY_BUILD_CLOSURE", "fortran"),
             self.define_from_variant("SINGULARITY_BUILD_PYTHON", "python"),
-            self.define_from_variant("SINGULARITY_BUILD_TESTS", "tests"),
+            self.define("SINGULARITY_BUILD_TESTS", self.run_tests),
             self.define(
                 "SINGULARITY_BUILD_SESAME2SPINER", "sesame" in self.spec.variants["build_extra"]
             ),
             self.define(
                 "SINGULARITY_TEST_SESAME",
-                ("sesame" in self.spec.variants["build_extra"] and "tests" in self.spec),
+                ("sesame" in self.spec.variants["build_extra"] and self.run_tests),
             ),
             self.define(
                 "SINGULARITY_BUILD_STELLARCOLLAPSE2SPINER",
                 "stellarcollapse" in self.spec.variants["build_extra"],
             ),
             self.define(
                 "SINGULARITY_TEST_STELLARCOLLAPSE2SPINER",
-                ("stellarcollapse" in self.spec.variants["build_extra"] and "tests" in self.spec),
-            ),
-            self.define(
-                "SINGULARITY_TEST_PYTHON", ("python" in self.spec and "tests" in self.spec)
+                ("stellarcollapse" in self.spec.variants["build_extra"] and self.run_tests),
             ),
+            self.define("SINGULARITY_TEST_PYTHON", ("python" in self.spec and self.run_tests)),
```

Comment:
Should this be ensuring the `python` variant is enabled?

```suggestion
            self.define("SINGULARITY_TEST_PYTHON", ("+python" in self.spec and self.run_tests)),
```

Reply:
Yes, good catch!